stdlib/Profile/src/Profile.jl

```diff
@@ -350,18 +350,18 @@ end
 
 function getdict!(dict::LineInfoDict, data::Vector{UInt})
     # we don't want metadata here as we're just looking up ips
-    unique_data_itr = Iterators.unique(has_meta(data) ? strip_meta(data) : data)
-    foreach(ip -> dict[UInt64(ip)] = StackFrame[], unique_data_itr)
-    @sync for ip in unique_data_itr
-        Threads.@spawn begin
+    unique_ips = unique(has_meta(data) ? strip_meta(data) : data)
+    chnl = Channel{Tuple{UInt64,Vector{StackFrame}}}(length(unique_ips)) do ch
+        Threads.@threads for ip in unique_ips
```

Comment:
This will hang if a thread is already running in non-yielding code. Can we at a minimum make this threading optional?

Reply:
Good point. Is there a tidy way to make a `@threads` optional?

Reply:
I have no idea, I was wondering the same thing when I wrote this. Maybe refactor the internals into its own function?

Reply:
Just pushed what appears to be non-blocking in this case